parser/prism/Translator.cc

```diff
@@ -1357,12 +1361,33 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
                 ast::Array::ENTRY_store argExprs;
                 argExprs.reserve(prismArgs.size());
                 for (auto &arg : args) {
+                    if (parser::NodeWithExpr::isa_node<parser::ForwardedRestArg>(arg.get())) {
```

Comment:
This is replicating the removal logic here:

https://github.com/sorbet/sorbet/blob/fb4fe38c4828cf9c054768413256b4982efe7d05/ast/desugar/Desugar.cc#L776-L779

Instead of removing elements after-the-fact, we just skip appending them in the first place.